modules/fileout_netcdf/FONcArrayStructure.cc

```diff
@@ -0,0 +1,220 @@
+ * We must flatten out the structures. To do this, we embed the name of the
+ * structure as part of the name of the children variables. For example,
+ * if the structure, called s1, contains an array called a1 and an int
+ * called i1, then two variables are created in the netcdf file called
+ * s1.a1 and s1.i1.
+ *
+ * @note This method only converts the variables that are to be sent. This keeps
+ * the convert() and write() methods below from operating on DAP variables
+ * that should not be sent.
+ *
+ * @param embed The parent names of this structure.
+ * @throws BESInternalError if there is a problem converting this
+ * structure
+ */
+void FONcArrayStructure::convert(vector<string> embed, bool _dap4, bool is_dap4_group){
+    set_is_dap4(_dap4);
+    FONcBaseType::convert(embed,_dap4,is_dap4_group);
+    embed.push_back(name());
+
+    auto as_v = dynamic_cast<Structure *>(_as->var());
+    if (!as_v) {
+        string s = (string) "File out netcdf, write an array of structure was passed a " + "variable that is not a structure";
+        throw BESInternalError(s, __FILE__, __LINE__);
+    }
+
+    if (false == isNetCDF4_ENHANCED()) {
+        throw BESInternalError("Fileout netcdf: We don't support array of structure for the classical model now. ",
+                               __FILE__, __LINE__);
+    }
+
+    Constructor::Vars_iter vi = as_v->var_begin();
+    Constructor::Vars_iter ve = as_v->var_end();
+    for (; vi != ve; vi++) {
+        BaseType *bt = *vi;
+        if (bt->send_p()) {
+            BESDEBUG("fonc", "FONcArrayStructure::convert - converting " << bt->name() << endl);
+
+            auto fsf = new FONcArrayStructureField(bt, _as);
+            _vars.push_back(fsf);
+            fsf->convert(embed,_dap4,is_dap4_group);
+        }
+    }
+
+}
+
+/** @brief Define the members of the array of structure in the netcdf file
+ *
+ * We still follow the current structure implementation to define the members of
+ * the structure to include the name of the structure in their name.
+ *
+ * @note This will call the FONcArrayStructureField's define() method for individual member.
+ * Because the FONcArrayStructure::convert() method above only
+ * builds a FONcArrayStructure for elements of the DAP Structure with send_p true,
+ * the code is protected from trying to operate on DAP variables that should
+ * not be sent. This is important because those variables likely don't contain
+ * any values!
+ *
+ * @param ncid The id of the netcdf file
+ * @throws BESInternalError if there is a problem defining the structure
+ */
+void FONcArrayStructure::define(int ncid)
+{
+
+    if (!d_defined) {
+        BESDEBUG("fonc", "FONcArrayStructure::define - defining " << d_varname << endl);
+        vector<FONcArrayStructureField *>::const_iterator i = _vars.begin();
+        vector<FONcArrayStructureField *>::const_iterator e = _vars.end();
+        for (; i != e; i++) {
```

Comment:
for (auto &fbt: _vars) {
    BESDEBUG("fonc", "defining " << fbt->name() << endl);
    fbt->define(ncid);
}

Reply:
Yes,  just copied the old way from FONcStructure.cc. Use the concise way.  Change the code.